rtree.go

```diff
@@ -739,9 +739,20 @@ func (tree *Rtree) nearestNeighbor(p Point, n *node, d float64, nearest Spatial)
 			}
 		}
 	} else {
-		branches, dists := sortEntries(p, n.entries)
-		branches = pruneEntries(p, branches, dists)
-		for _, e := range branches {
+		minMinMaxDist := math.MaxFloat64
```

Comment:
it's been a long time since I wrote this, I think I took it nearly verbatim from the paper, and so the correctness of the change here is not immediately obvious to me - can you add a comment explaining the approach?